tests/helpers/test_script_variables.py

```diff
@@ -0,0 +1,112 @@
+"""Test script variables."""
+import pytest
+
+from homeassistant.helpers import config_validation as cv, template
+
+
+async def test_static_vars():
+    """Test static vars."""
+    orig = {"hello": "world"}
+    var = cv.SCRIPT_VARIABLES_SCHEMA(orig)
+    rendered = var.async_render(None, None)
+    assert rendered is not orig
+    assert rendered == orig
+
+
+async def test_static_vars_run_args():
+    """Test static vars."""
+    orig = {"hello": "world"}
+    orig_copy = dict(orig)
+    var = cv.SCRIPT_VARIABLES_SCHEMA(orig)
+    rendered = var.async_render(None, {"hello": "override", "run": "var"})
+    assert rendered == {"hello": "override", "run": "var"}
+    # Make sure we don't change original vars
+    assert orig == orig_copy
+
+
+async def test_static_vars_no_default():
+    """Test static vars."""
+    orig = {"hello": "world"}
+    var = cv.SCRIPT_VARIABLES_SCHEMA(orig)
+    rendered = var.async_render(None, None, render_as_defaults=False)
+    assert rendered is not orig
+    assert rendered == orig
+
+
+async def test_static_vars_run_args_no_default():
+    """Test static vars."""
+    orig = {"hello": "world"}
+    orig_copy = dict(orig)
+    var = cv.SCRIPT_VARIABLES_SCHEMA(orig)
+    rendered = var.async_render(
+        None, {"hello": "override", "run": "var"}, render_as_defaults=False
+    )
+    assert rendered == {"hello": "world", "run": "var"}
+    # Make sure we don't change original vars
+    assert orig == orig_copy
+
+
+async def test_template_vars(hass):
+    """Test template vars."""
+    var = cv.SCRIPT_VARIABLES_SCHEMA({"hello": "{{ 1 + 1 }}"})
+    rendered = var.async_render(hass, None)
+    assert rendered == {"hello": "2"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)